src/inmanta/protocol/openapi/converter.py

```diff
@@ -180,18 +186,24 @@ def _handle_list(self, type_annotation: Type) -> Schema:
         return Schema(type="array", items=self.get_openapi_type(list_member_type[0]))
 
     def get_openapi_type(self, type_annotation: Type) -> Schema:
-
-        if typing_inspect.is_optional_type(type_annotation):
+        if (
+            typing_inspect.is_optional_type(type_annotation)
+            and len(typing_inspect.get_args(type_annotation, evaluate=True)) == 2
+        ):
             return self._handle_optional_type(type_annotation)
-        elif typing_inspect.is_union_type(type_annotation):
+        if typing_inspect.is_union_type(type_annotation):
             return self._handle_union_type(type_annotation)
         elif inspect.isclass(type_annotation) and issubclass(type_annotation, BaseModel):
             return self._handle_pydantic_model(type_annotation)
         elif inspect.isclass(type_annotation) and issubclass(type_annotation, Enum):
             return self._handle_enums(type_annotation)
-        elif typing_inspect.get_origin(type_annotation) == dict or typing_inspect.get_origin(type_annotation) == typing.Dict:
+        elif inspect.isclass(typing_inspect.get_origin(type_annotation)) and issubclass(
+            typing_inspect.get_origin(type_annotation), typing.Mapping
```

Comment:
please factor out 'typing_inspect.get_origin(type_annotation)' to a local variable, 

(I know this code is not that performance sensitive, but it is used soooo many time),